syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.walmartlabs.concord.rpc";
option java_outer_classname = "ServerProto";
option objc_class_prefix = "HLW";

package agent_api;

import "google/protobuf/empty.proto";
import "google/protobuf/any.proto";
import "google/protobuf/timestamp.proto";

// SECRET STORE SERVICE (DEPRECATED)

service TSecretReaderService {

    rpc fetch(TFetchSecretRequest) returns (TFetchSecretResponse);
    rpc decrypt(TDecryptRequest) returns (TDecryptResponse);
}

message TFetchSecretRequest {
    string instanceId = 1;
    string secretName = 2;
    string password = 3;
    string orgName = 4;
}

enum TSecretType {
    KEY_PAIR = 0;
    USERNAME_PASSWORD = 1;
    DATA = 2;
}

message TFetchSecretResponse {
    string id = 1;
    TFetchSecretStatus status = 2;
    TSecretType type = 3;
    bytes data = 4;
}

enum TFetchSecretStatus {
    SECRET_FOUND = 0;
    SECRET_NOT_FOUND = 1;
    SECRET_NOT_AUTHORIZED = 2;
    SECRET_STORE_ERROR = 3;
}

message TDecryptRequest {
    string instanceId = 1;
    bytes data = 2;
}

enum TDecryptStatus {
    KEY_FOUND = 0;
    KEY_NOT_FOUND = 1;
    INVALID_DATA = 2;
}

message TDecryptResponse {
    TDecryptStatus status = 1;
    bytes data = 2;
}

// EVENT SERVICE

service TEventService {
    rpc onEvent(TEventRequest) returns (google.protobuf.Empty);
}

enum TEventType {
    PROCESS_ELEMENT = 0;
    ANSIBLE_EVENT = 1;
}

message TEventRequest {
    string instanceId = 1;
    TEventType type = 2;
    google.protobuf.Timestamp date = 3;
    bytes data = 4;
}